allofplos/article_class.py

```diff
@@ -858,6 +858,23 @@ def url(self):
         """
         return self.get_url(plos_network=self.plos_network)
 
+    @property
+    def taxonomy(self):
+        """Taxonomy information
+        """
+        tag_path_elements = ('/',
+                             'article',
+                             'front',
+                             'article-meta',
+                             'article-categories')
+        e_list = self.get_element_xpath(tag_path_elements=tag_path_elements)
+        taxonomy_item = []
+        for subj in e_list[0].getchildren():
+            if subj.values()[0] == 'Discipline-v3':
```

Comment:
I wonder if we should capture also the taxonomy heading (Discipline, Discipline-V2, Discipline-V3). I could include in the touple if it provides some value.
What do you think?

Reply:
I think including version number is a great idea. What if you made it a dict, where the keys are the version numbers/headers and the values are the lists of tuples? That matches how Article() handles contributor info that can come from different sources.

Reply:
also in the docstring, we probably want to link to https://github.com/PLOS/plos-thesaurus so people can grab the entire taxonomy if they wish.

Reply:
I found a new case: 10.1371/journal.pone.0015450
it has two taxonomy heading not in the list: 'Prion diseases' and 'Discipline'
Will add them

Reply:
This one was an "=" in the heading ("Discipline-v2="):
10.1371/journal.pone.0016652

Reply:
This is the list of all headers, there are several spellings/variations for discipline.
Instead of check them, I will grab the header and use it as a dictionary key. 

Predictive toxicology
Algorithms
discipline
Cellular stress responses
Infectious disease modeling
Molecular genetics
Mouse
;Discipline-v2
Disclipline-v2
Nuclear receptor signaling
 Discipline-v2
heading
Trees
Discipline-b2
Classification
Discipline-v2\
Diipline-v2
Numerical analysis
Discipline-v2 
Disclipine-v2
Behavioral ecology
Smoking related disorders
Experimental psychology
Discipline-v2&
discipline-v2
Microbial mutation
Discipline
Disipline
Learning
Category
Discipline v-2
Environmental impacts
Discipline-V2
System Taxonomy
Discipline-v2=
Immune cells
Discipline-v
Small molecules
Oncogenic signaling
         Article
Biology
iscipline-v2
Article
Dicipline
DNA modification
Discipline-v3
Phenotypes
Discipline-v2
Human geography
Prion diseases
Sensory perception
Discipline--v2
Statistical mechanics
Cognition